sample_formatted_code.do_not_merge.c

```diff
@@ -0,0 +1,562 @@
+
+    memset(fileext, 0x00, sizeof(DetectFileextData));
+
+    if (DetectContentDataParse("fileext", str, &fileext->ext, &fileext->len,
+                &fileext->flags) == -1) {
+        goto error;
+    }
+
+    return fileext;
+
+error:
+    if (fileext != NULL)
+        DetectFileextFree(fileext);
+    return NULL;
+}
+
+int goto_style_nested()
+{
+    // IndentGotoLabels
+    if (foo()) {
+    label1:
+        bar();
+    }
+
+label2:
+    return 1;
+}
+
+//--- ternary style ---
+// BreakBeforeTernaryOperators - whether ? and : are on next line
+void ternary_op_style()
+{
+    // fits on one line
+    float droppy = a > 0 ? a * 100 : 0;
+
+    // split across lines - Continuation indent based on '=' fits line
+    float drop_percent = likely(ptv->last_stats64.ps_recv > 0)
+                                 ? (((float)ptv->last_stats64.ps_drop) /
+                                           (float)ptv->last_stats64.ps_recv) *
+                                           100
+                                 : 0;
+
+    // split across lines - Continuation indent based on '=' would be too long,
+    // uses normal continuation indent based on start of line
+    float drop_percent_a_bit_longer =
+            likely(ptv->last_stats64.ps_recv > 0)
+                    ? (((float)ptv->last_stats64.ps_drop) /
+                              (float)ptv->last_stats64.ps_recv) *
+                              100
+                    : 0;
+}
+
+//--- enum style ---
+// clang 11: AllowShortEnumsOnASingleLine
+// clang < 11:
+// - merges short enums on one line if BraceWrapping: AfterEnum: false
+// - one-value-by-line if BraceWrapping: AfterEnum: true
+enum Gugus { bla, bli, blu };
+
+enum ThisIsTooLongForOneLine {
+    blablablablablablablabla,
+    blibliblibliblibliblibli,
+    blublublublublublublublu
+};
+
+enum { A, B } myEnum;
+
+// trailing comma forces one-value-by-line
+enum {
+    NFS_DECODER_EVENT_EMPTY_MESSAGE,
+};
+
+//--- union style ---
+typedef union {
+    int gugus;
+} Bla;
+
+union bla {
+    int gugus;
+};
+
+// --- struct style ---
+struct bla {
+    int gugus;
+};
+
+struct bla_ {
+    int gugus;
+} Bla;
+
+typedef struct bla_ {
+    int gugus;
+} Bla;
+
+//--- Alignment ---
+struct bla {
+    // AlignConsecutiveDeclarations
+    // AlignTrailingComments
+    int a;       /* comment */
+    unsigned bb; /* comment */
+    int *ccc;    /* comment */
+};
+
+// pointers
+// DerivePointerAlignment
+// PointerAlignment
+void *ptr;
+void f(int *a, const char *b);
+void (*foo)(int *);
+
+void alignment()
+{
+    // multiple consecutive vars (comments and vars can be aligned)
+    // AlignConsecutiveAssignments
+    // AlignConsecutiveDeclarations
+    // AlignTrailingComments
+    int a = 13;           /* comment */
+    int32_t abc = 1312;   /* comment */
+    int abcdefghikl = 13; /* comment */
+
+    // AlwaysBreakBeforeMultilineStrings
+    aaaa = "bbbb"
+           "ccc";
+
+    //--- variable init continuation behaviour
```

Comment:
I left this out in code-style.rst. Seemed too mundane of a detail. Only useful for initial choice of setting, imho.